src/scripts/modal.js

```diff
@@ -9,18 +9,29 @@ function modal() {
   function close() {
     modalContent.classList.remove("modal--show");
     heroContent.classList.remove("blur");
+
+    checkOfferBtn.disabled = false;
+
+    document.removeEventListener("click", closeOutside, true);
   }
-  checkOfferBtn.addEventListener("click", () => {
+
+  function open() {
     modalContent.classList.add("modal--show");
     heroContent.classList.add("blur");
-  });
+
+    checkOfferBtn.disabled = true;
+
+    document.addEventListener("click", closeOutside, true);
+  }
+
+  function closeOutside(e) {
+    modalContent.contains(e.target) ? "" : close();
```

Comment:
No need for an empty string. It should be an if statement:
```ts
    if (!modalContent.contains(e.target)) {
      close();
    }
```

Reply:
changed